cart/infrastructure/defaultCartBehaviour.go

```diff
@@ -97,96 +119,80 @@ func (cob *DefaultCartBehaviour) Complete(ctx context.Context, cart *domaincart.
 	if err != nil {
 		return nil, nil, err
 	}
+
 	return cart, nil, nil
 }
 
 // Restore supplied cart (implements CompleteBehaviour)
 func (cob *DefaultCartBehaviour) Restore(ctx context.Context, cart *domaincart.Cart) (*domaincart.Cart, domaincart.DeferEvents, error) {
 	newCart, err := cart.Clone()
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, errors.Wrap(err, "cart.infrastructure.DefaultCartBehaviour: error on cloning cart")
```

Comment:
please use `fmt.Errorf("there is a issue: %w", err)`, and drop the `"github.com/pkg/errors"` import since that package is deprecated.